pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -102,7 +102,63 @@ func TestDevicePluginReRegistration(t *testing.T) {
 	err = p3.Start()
 	require.NoError(t, err)
 	atomic.StoreInt32(&expCallbackCount, 3)
-	p3.Register(socketName, testResourceName)
+	p3.Register(socketName, testResourceName, false)
+	// Wait for the second callback to be issued.
+	<-callbackChan
+
+	devices3 := m.Devices()
+	require.Equal(t, 1, len(devices3[testResourceName]), "Devices of plugin previously registered should be removed.")
+	p2.Stop()
+	p3.Stop()
+	cleanup(t, m, p1)
+	close(callbackChan)
+}
+
+func TestDevicePluginReRegistrationWithDevInitTrue(t *testing.T) {
```

Comment:
Can you add a comment on why you are testing this ?
AFAIK the ReRegister bug affects the registration system not the Allocate / Init system :)

Reply:
just to ensure that  initRequired has not broken registration, since it involes registration api change as well